examples/displayio_layout_gridlayout_dividers.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-FileCopyrightText: 2021 Tim C, written for Adafruit Industries
+#
+# SPDX-License-Identifier: MIT
+"""
+Illustrate how to use divider lines with GridLayout
+"""
+import board
+import displayio
+import terminalio
+from adafruit_display_text import label
+from adafruit_displayio_layout.layouts.grid_layout import GridLayout
+
+# use built in display (PyPortal, PyGamer, PyBadge, CLUE, etc.)
+# see guide for setting up external displays (TFT / OLED breakouts, RGB matrices, etc.)
+# https://learn.adafruit.com/circuitpython-display-support-using-displayio/display-and-display-bus
+display = board.DISPLAY
+
+# Make the display context
+main_group = displayio.Group()
+display.show(main_group)
+
+layout = GridLayout(
+    x=10,
+    y=10,
+    width=200,
+    height=100,
+    grid_size=(2, 2),
+    cell_padding=8,
+    divider_lines=True,  # divider lines around every cell
+)
+_labels = []
+
+_labels.append(
+    label.Label(
+        terminalio.FONT, scale=2, x=0, y=0, text="Hello", background_color=0x770077
+    )
+)
+layout.add_content(_labels[0], grid_position=(0, 0), cell_size=(1, 1))
+_labels.append(
+    label.Label(
+        terminalio.FONT, scale=2, x=0, y=0, text="World", background_color=0x007700
+    )
+)
+layout.add_content(_labels[1], grid_position=(1, 0), cell_size=(1, 1))
+_labels.append(label.Label(terminalio.FONT, scale=2, x=0, y=0, text="Hello"))
+layout.add_content(_labels[2], grid_position=(0, 1), cell_size=(1, 1))
+_labels.append(label.Label(terminalio.FONT, scale=2, x=0, y=0, text="Grid"))
+layout.add_content(_labels[3], grid_position=(1, 1), cell_size=(1, 1))
+
+main_group.append(layout)
+
+other_layout = GridLayout(
+    x=10,
+    y=120,
+    width=140,
+    height=80,
+    grid_size=(2, 3),
+    cell_padding=4,
+    divider_lines=True,
+    h_divider_line_rows=(1, 2),  # Lines before rows 1 and 2
+    v_divider_line_cols=(),  # No vertical divider lines
```

Comment:
I wanted to break the API down into two main use cases, the simplest being on/off for divider lines to get drawn at all. The user can enable them by passing `divider_lines=True` in the constructor. This will cause it to draw all available lines around every cell.

The second and slightly more advanced use case is the user wants to specify which cells and in which orientation to draw lines some of the lines. In this case with the current API the passes `divider_lines=True` and then passes the desired configurations with the `h_divider_line_rows` and `v_divider_line_cols` arguments. If either of those arguments is omitted then it reverts to the standard `divider_lines=True` behavior which is to draw all available lines. So currently passing the empty tuple is necessary in order to specify that it should not draw any lines for the given orientation. 

Maybe it would be simpler if the user passes either `h_divider_line_rows` or `v_divider_line_cols` then they do not need to also pass `divider_lines=True` I think that would allow the logic to work opposite how it does currently. It could start from drawing no lines, and only draw the ones specified by `h_divider_line_rows` and `v_divider_line_cols` since `divider_lines=True` would not be passed it would not draw all available lines if either of those is omitted. 

I can look into this further and work on those changes tonight. 

It seems I may have unintentionally changed the positioning behavior with this PR as well. I noticed just now that running the example in the MacroPad repo results in the labels getting shifted further down than they should be and causing the bottom row to get drawn half way off of the screen. I will try to resolve this issue when I work on it as well.

Reply:
I noticed the shift and forgot to mention it! I didn't even think about the fact that it was an inadvertent change. Indeed, I had to alter my y position by 7 pixels to get everything back in position.